### PR TITLE
uds.py: add basic AuthenticationService and more details to SecuredDataTransmission from ISO 14229-1:2020

### PR DESCRIPTION
**Pull request by sigttou**
[UDS] ADD: more details to SecuredDataTransmission
[UDS] ADD: Authentication Service
[TEST] Add: Basic unit tests for Auth and SDT

### CODE REVIEW

**Comment by polydroi** [comment]
Thanks for this PR. Could you please add a description for service 0x69.

File: scapy/contrib/automotive/uds.py [hunk 1]
```diff
@@ -48,6 +48,7 @@ class UDS(ISOTP):
          0x24: 'ReadScalingDataByIdentifier',
          0x27: 'SecurityAccess',
          0x28: 'CommunicationControl',
+         0x29: 'Authentication',
          0x2A: 'ReadDataPeriodicIdentifier',
```
Comment (polydroi) [inline scapy/contrib/automotive/uds.py:52]: Missing service description for 0x69

**Comment by polydroi** [comment]
By looking at the UDS_AUTH class, I'm wondering if it would make sense to also provide a customized answers function for UDS_AUTHPR which compares the subfunction.

File: scapy/contrib/automotive/uds.py (hunk 1 above)
Comment (sigttou) [reply scapy/contrib/automotive/uds.py:52]: Thanks, missed that one - added.

File: scapy/contrib/automotive/uds.py [hunk 2]
```diff
@@ -273,6 +275,156 @@ def answers(self, other):
...
+                         lambda pkt: pkt.subFunction == 0x05),
+        ConditionalField(XStrLenField('neededAdditionalParameter', b"",
+                                      length_from=lambda p:
+                                      p.lengthOfNeededAdditionalParameter),
+                         lambda pkt: pkt.subFunction == 0x05),
+    ]
+
+    def answers(self, other):
+        return isinstance(other, UDS_AUTH) \
+            and other.subFunction == self.subFunction
```
Comment (sigttou) [inline scapy/contrib/automotive/uds.py:422]: @polydroi  - the check for the subFunction is in there

**Comment by sigttou** [comment]
As I found the time today during applying the feedback, I've also added the missing fields from Auth Service (get rid of dataRecord).

**Comment by polydroi** [comment]
Thanks for this additional changes. One more small request from my side. Could you please also add some build tests for both packets. All dissection tests look good so far, but I would like to ensure that the length_from works as expected during a build of a packet.

**Comment by sigttou** [comment]
You were right, the length_from with a default value 0 doesn't work as expected, it should be None. As I just copied from RFT I also made the fix there and wrote build tests for it.

**Comment by polybassa** [comment]
BTW: @polydroi is my account, I use on Android.

**Comment by polybassa** [comment]
Looks all good to me. @guedou or @gpotter2 Could you please enable CI for this PR?

**Review by polybassa — APPROVED** [review]

**Review by guedou — APPROVED** [review]

**Pull request merged**